### PR TITLE
Replace log2 and log10 with Julia ports.

### PR DESCRIPTION
**Pull request by pkofod**
- [x] fix license
- [x] add tests
- [x] consider moving all sorts of helper functions (highword, zero_out_lowword) up in the file tree so they're not scattered around in various files.

### CODE REVIEW

File: base/special/log.jl [hunk 1]
```diff
@@ -395,3 +395,250 @@ for f in (:log,:log1p)
...
+const lg2 = 0.40000972f0
+const lg3 = 0.28498787f0
+const lg4 = 0.24279079f0
+
+_log1p_t1(w::Float32) = @horner(w, lg2, lg4)
+_log1p_t1(w::Float64) = @horner(w, Lg2, Lg4, Lg6)
+_log1p_t2(w::Float32) = @horner(w, lg1, lg3)
+_log1p_t2(w::Float64) = @horner(w, Lg1, Lg3, Lg5, Lg7)
+"""
+    _log1p(f::Union{Float32, Float64})
```
Comment (giordano) [inline base/special/log.jl]: This docstring mentions `_log1p` but the method is called `k_log1p`.

**Comment by ViralBShah** [comment]
Bump - can we get this in?

**Comment by pkofod** [comment]
I’m not in possession of a computer I can dev on until next week

**Comment by ViralBShah** [comment]
Bump - in order to get this (and the rest of the libm) things merged, in case we are close enough.

**Comment by ViralBShah** [comment]
Bump - would be nice to revive this.

**Comment by pkofod** [comment]
If the kids cooperate (by falling asleep instead of screaming for random things), I have some time tonight to add the benchmarks and tests.

**Comment by pkofod** [comment]
Seems to run a bit faster but I only checked it against 1.0.2 prebuilt binaries, but I did a stupid thing and my system is now building from scratch. >_>

**Comment by pkofod** [comment]
Okay, so I had a look at the commits, and the regression is likely due to a branch in the range reduction that is no longer there, I'm talking about the `log_small` values in the original commit. What do you say @simonbyrne ? I mean, the benchmarks specifically test that branch, but it's not clear how important it is.

File: base/special/log.jl [hunk 2]
```diff
@@ -395,3 +398,139 @@ for f in (:log,:log1p)
         ($f)(x::Real) = ($f)(float(x))
     end
 end
+
+# Below is line are log10 and log2 ports from Openlibm. The following copyright
```
Comment (Peiffap) [inline base/special/log.jl]:
```suggestion
# Below this line are log10 and log2 ports from Openlibm. The following copyright
```

**Comment by ViralBShah** [comment]
What do we need to get this merged?

**Comment by pkofod** [comment]
I've not sure if I ever got around to rebuilding as mentioned above, but given some recent oddities benchmarking these things, I think we should benchmark them once more, and I'd love to have some knowledgeable people looking over my shoulder to make sure that there are no regressions.

File: base/special/log.jl [hunk 3]
```diff
@@ -395,3 +398,139 @@ for f in (:log,:log1p)
...
+# ====================================================
+# Copyright (C) 1993 by Sun Microsystems, Inc. All rights reserved.
+#
+# Developed at SunSoft, a Sun Microsystems, Inc. business.
+# Permission to use, copy, modify, and distribute this
+# software is freely granted, provided that this notice
+# is preserved.
+# ====================================================
+
+# Constants and utilities
```
Comment (ggggggggg) [inline base/special/log.jl]: is this an orphan comment that could be removed?

**Comment by aminya** [comment]
I tested this, and it works. Is there a reason that we have not merged it?
My test branch:
https://github.com/aminya/julia/tree/log2log10

**Review by aminya — APPROVED** [review]
I tested this, and it works. Is there a reason that we have not merged it?
My test branch:
https://github.com/aminya/julia/tree/log2log10

**Comment by StefanKarpinski** [comment]
Bump, would be lovely to get this out of WIP...

**Comment by pkofod** [comment]
WIP is probably also no longer descriptive...  I think the missing step here is proper benchmarking.

**Comment by pkofod** [comment]
what have i done :scream:

**Comment by ViralBShah** [comment]
I am wondering that if we are redoing this, do we prefer getting the Tang algorithm that is in libm.jl?

**Comment by pkofod** [comment]
> I am wondering that if we are redoing this, do we prefer getting the Tang algorithm that is in libm.jl?

Good question. Since this is essentially sure, I'm not sure it's worth it. I mean we can always try that one after. The main thing to be done here is to test for regressions, but I've not felt confident doing benchmarks on these types of functions in Julia for a while. It's so hard to know if things get inlined/constantfolded/etc.

**Comment by oscardssmith** [comment]
https://github.com/JuliaLang/julia/pull/39556 supersedes.

**Pull request closed without merging**